.tekton/on-cm-runner.yaml

```diff
@@ -0,0 +1,274 @@
+              volumeMounts:
+                - name: google-creds-volume
+                  mountPath: /etc/secrets/google
+                  readOnly: true
+
+              env:
+                - name: GOOGLE_SERVICE_ACCOUNT_FILE
+                  value: "/etc/secrets/google/credentials.json"
+                - name: GSHEETS_MODE
+                  value: "both"
+                - name: GSHEETS_TAG
+                  value: "Ver1"
+                - name: GSHEETS_INPUT_SHEET_ID
+                  valueFrom:
+                    secretKeyRef:
+                      name: google-sheets-secrets
+                      key: input_sheet_id
+                - name: GSHEETS_OUTPUT_SHEET_ID
+                  valueFrom:
+                    secretKeyRef:
+                      name: google-sheets-secrets
+                      key: output_sheet_id
+
+              script: |
+                #!/bin/bash
+                set -e 
+                
+                # 1. Construct the Tag Name
+                SHORT_HASH=$(echo $(params.CURRENT_REVISION) | cut -c1-7)
+                TAG_NAME="pr-$(params.PR_NUMBER)-${SHORT_HASH}"
+                
+                echo "--- STARTING AUTOMATION TESTS ---"
+                echo "Generated Tag: $TAG_NAME"
+                
+                # 2. Run Python script with the tag
+                python3 src/vulnerability_main_automation.py --gsheets-tag "$TAG_NAME"
+                
+                echo "--- TESTS FINISHED SUCCESSFULLY ---"
+
+            # -------------------------------------------------------
+            # STEP B: Tag the MAIN Repository
+            # -------------------------------------------------------
+            - name: git-tag-main-repo
+              image: alpine/git
+              workingDir: $(workspaces.source.path)
+              script: |
+                #!/bin/sh
+                set -e
+                
+                SHORT_HASH=$(echo $(params.CURRENT_REVISION) | cut -c1-7)
+                TAG_NAME="pr-$(params.PR_NUMBER)-${SHORT_HASH}"
+                
+                echo "--- Configuring Git for Main Repo ---"
+                git config --global user.email "tekton-bot@redhat.com"
+                git config --global user.name "Tekton Pipeline"
+                
+                # Auth Setup using the mounted secret
+                if [ -f "/workspace/basic-auth/password" ]; then
+                   GIT_PASSWORD=$(cat /workspace/basic-auth/password)
+                   GIT_USERNAME=$(cat /workspace/basic-auth/username)
+                   ORIGIN_URL=$(git remote get-url origin)
+                   # Strip protocol and add auth
+                   CLEAN_URL=${ORIGIN_URL#*//}
+                   git remote set-url origin "https://${GIT_USERNAME}:${GIT_PASSWORD}@${CLEAN_URL}"
+                fi
+
+                echo "--- Pushing Tag '$TAG_NAME' to Main Repo ---"
+                git tag -f -a "$TAG_NAME" -m "Tests Passed for PR #$(params.PR_NUMBER)"
```

Comment:
@RedTanny  Did you notice that this tag is a tag on the current repository ( just pointing to the HEAD branch of the current PR, and the tag from the next step  ( STEP C) is tag with the same name  that is created automatically in the  exploitiq-cm-automation repo using the github CLI release command?
These two are unrelated...., and because this tag doesn't contain anything but the head pr branch tip commit, why does it needed here?